ietf/meeting/views.py

```diff
@@ -2568,6 +2569,56 @@ def add_session_drafts(request, session_id, num):
                     'form': form,
                   })
 
+class SessionRecordingsForm(forms.Form):
+    title = forms.CharField(max_length=255)
+    url = forms.URLField(label="Link to recording (YouTube only)")
+
+    def clean_url(self):
+        url = self.cleaned_data['url']
+        parsed_url = urlparse(url)
+        if parsed_url.hostname not in YOUTUBE_DOMAINS:
+            raise forms.ValidationError("Must be a YouTube URL")
+        return url
+
+
+def add_session_recordings(request, session_id, num):
+    # num is redundant, but we're dragging it along an artifact of where we are in the current URL structure
+    session = get_object_or_404(Session,pk=session_id)
+    if not session.can_manage_materials(request.user):
+        raise Http404
+    if session.is_material_submission_cutoff() and not has_role(request.user, "Secretariat"):
+        raise Http404
+
+    session_number = None
+    sessions = get_sessions(session.meeting.number,session.group.acronym)
+    initial = {
+        'title': f"Video recording for {session.group.acronym} on {session.meeting.date.strftime('%b-%d-%Y at %H:%M:%S')}"
+    }
```

Comment:
To be more concrete about my comments elsewhere, I think something like this is what is needed. The `assertion()` call is not bulletproof, there'll still be a 500 if this comes up.
```suggestion
    timeslot = session.official_timeslotassignment()
    assertion("timeslot is not None")
    initial = {
        'title': f"Video recording for {session.group.acronym} on {timeslot.utc_start_time().strftime('%b-%d-%Y at %H:%M:%S')}"
    }
```

Reply:
Noticed I botched the `timeslotassignment.timeslot` step - glad you caught it!